sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -76,6 +76,8 @@
     /**
      * Package-private constructor for use by {@link BlobClientBuilder}.
      *
+     * <p>Blob name is encoded to UTF-8 using the {@link com.azure.storage.common.Utility#urlEncode(String)} method.</p>
```

Comment:
I don't think these comments on the constructors are accurate because the encoding happens in the builder. It seems like it's more accurate to say these constructors expect that the blobName is already properly encoded.

Reply:
Sounds good, I'll change the comments on the constructors.

Reply:
I thought of adding said comments wherever a `blobName` or URL/endpoint is passed as an argument to generate a blob client so that the user knew what would happen to it. In hindsight, I think we could omit these comments wherever the `blobName` is not directly altered. What do you think?

As a side note, there are some cases where the `blobName` is set in the constructor, such as `BlobAsyncClientBase`, which is used to create our specialized blob clients.